resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -108,15 +106,16 @@ private[spark] class Client(
         .withNewMetadata()
           .withName(secretName)
           .endMetadata()
-        .withData(Map((SUBMISSION_SERVER_SECRET_NAME, secretBase64String)).asJava)
+        .withData(Map((SUBMISSION_APP_SECRET_NAME, secretBase64String)).asJava)
         .withType("Opaque")
         .done()
       val (sslEnvs, sslVolumes, sslVolumeMounts, sslSecrets) = configureSsl(kubernetesClient,
-        driverLaunchSslOptions,
+        driverSubmitSslOptions,
         isKeyStoreLocalFile)
       try {
         val driverKubernetesSelectors = (Map(
-            DRIVER_LAUNCHER_SELECTOR_LABEL -> driverLauncherSelectorValue,
+            SPARK_DRIVER_LABEL -> kubernetesAppId,
+            SPARK_APP_ID_LABEL -> kubernetesAppId,
```

Comment:
The first label indicates that this is a spark driver as opposed to a Spark executor. Thus when trying to match labels for the driver you want to pick up only the pods with the driver label (and not the executor label, for example)